pyaedt/edb_grpc/core/edb_data/layer_data.py

```diff
@@ -0,0 +1,1753 @@
+            elif layer_type == self.layer_types.SIwaveHFSSSolverRegions:
+                return 16
+            elif layer_type == self.layer_types.OutlineLayer:
+                return 18
+        elif isinstance(layer_type, int):
+            return layer_type
+
+    @stackup_mode.setter
+    def stackup_mode(self, value):
+        if value == 0 or value == self.layer_collection_mode.Laminate:
+            self.layer_collection.SetMode(self.layer_collection_mode.Laminate)
+        elif value == 1 or value == self.layer_collection_mode.Overlapping:
+            self.layer_collection.SetMode(self.layer_collection_mode.Overlapping)
+        elif value == 2 or value == self.layer_collection_mode.MultiZone:
+            self.layer_collection.SetMode(self.layer_collection_mode.MultiZone)
+
+    @pyaedt_function_handler()
+    def _update_edb_objects(self):
+        self._edb_object = OrderedDict({})
+        layers = self.edb_layers
+        for i in range(len(layers)):
+            self._edb_object[layers[i].GetName()] = EDBLayer(layers[i], self)
+        return True
+
+    @pyaedt_function_handler()
+    def _update_stackup(self):
+        self._active_layout.SetLayerCollection(self.edb_layer_collection)
+        self._update_edb_objects()
+        return True
+
+    @pyaedt_function_handler()
+    def insert_layer_above(
+        self,
+        layer_name,
+        base_layer,
+        material="copper",
+        fillMaterial="",
+        thickness="35um",
+        layerType=0,
+        negative_layer=False,
+        roughness_enabled=False,
+        etch_factor=None,
+    ):
+        """Insert a layer above the specified base layer.
+
+        Parameters
+        ----------
+        layer_name : str
+            Name of the layer to add.
+        base_layer : str
+            Name of the layer after which to add the new layer.
+            The default is ``None``.
+        material : str, optional
+            Name of the material. The default is ``"copper"``.
+        fillMaterial : str, optional
+            Name of the fill material. The default is ``""``.)
+        thickness : str, optional
+            Thickness value, including units. The default is ``"35um"``.
+        layerType :
+            Type of the layer. The default is ``0``
+            ``0``: Signal layer.
+            ``1``: Dielectric layer.
+            ``2``: Conducting plane layer.
+            ``3``: Airline layer.
+            ``4``: Error layer.
+            ``5``: Symbol layer.
+            ``6``: Measure layer.
+            ``8``: Assembly layer.
+            ``9``: Silkscreen layer.
+            ``10``: Solder Mask layer.
+            ``11``: Solder Paste layer.
+        negative_layer : bool, optional
+            ``True`` when negative, ``False`` otherwise.
+        roughness_enabled : bool, optional
+            ``True`` if the layer has roughness, ``False`` otherwise.
+        etch_factor : optional
+            Etch value if any. The default is ``None``.
```

Comment:
```suggestion
            Type of the layer. The default is ``0``. Options are:
            
            - ``0``: Signal layer
            - ``1``: Dielectric layer
            - ``2``: Conducting plane layer
            - ``3``: Airline layer
            -``4``: Error layer
            - ``5``: Symbol layer
            ``6``: Measure layer.
            ``8``: Assembly layer.
            ``9``: Silkscreen layer.
            ``10``: Solder Mask layer.
            ``11``: Solder Paste layer.
        negative_layer : bool, optional
            ``True`` when negative, ``False`` otherwise.
        roughness_enabled : bool, optional
            ``True`` if the layer has roughness, ``False`` otherwise.
        etch_factor : optional
            Etch value if any. The default is ``None``.
```